### PR TITLE
Frontend/view profile

### PR DESCRIPTION
**Pull request by jennifer-lu**
## Notion ticket link

[View profile + logout drop down from nav bar](https://www.notion.so/uwblueprintexecs/View-profile-logout-drop-down-from-nav-bar-7efd92729fb147d0b935e6b3d81b0344)

## Implementation description

- Added user profile modal using Chakra UI Modal
- Profile and sign out links are text with `onClick` for consistent appearance
- Icons from `react-icons/bi` most closely matched those in Figma

## Steps to test

1. Sign in as `planetread+angelamerkel@uwblueprint.org`
2. Click on user icon in nav bar
3. Modal should say Angela Merkel Admin with option to sign out
4. Click away from the modal to close it
5. Reopen the modal and sign out
6. Sign in as `planetread+kevindebruyne@uwblueprint.org`
7. Click on user icon in nav bar
8. Modal should say Kevin De Bruyne Translator & Reviewer with options view profile and sign out
9. Click on profile to be redirected to user page
10. Go back
11. Reopen the modal and sign out
12. Sign in as `planetread+carlsagan@uwblueprint.org` (for me Carl is broken so I used `planetread+miroslavklose@uwblueprint.org` but removed his reviewer status using mysql)
13. Click on user icon in nav bar
14. Modal should say Carl Sagan Translator (or Miroslav Klose Translator) with options view profile and sign out
15. Click on profile to be redirected to user page
16. Go back
17. Reopen the modal and sign out

## What should reviewers focus on?

- Do links work
- Is code readable enough
- Is appearance good

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/navigation/UserModal.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    <Modal
+      isOpen={showUser}
+      onClose={onClose}
+      motionPreset="slideInBottom"
+      size="xs"
+    >
+      <ModalOverlay />
+      <ModalContent marginTop="30vh">
+        <Box
+          backgroundColor="blue.50"
```
Comment (gaoxk) [inline frontend/src/components/navigation/UserModal.tsx:52]: nit: alphabetize these props

Comment (gaoxk) [reply frontend/src/components/navigation/UserModal.tsx:52]: a few other components have this nit too

File: frontend/src/components/navigation/Header.tsx [hunk 2]
```diff
@@ -1,7 +1,10 @@
-import React from "react";
-import { Button, Flex, Heading, Image } from "@chakra-ui/react";
+import React, { useContext, useState } from "react";
+import { Button, Flex, Heading, IconButton, Image } from "@chakra-ui/react";
+import { Icon } from "@chakra-ui/icon";
+import { BiUserCircle } from "react-icons/bi";
```
Comment (gaoxk) [inline frontend/src/components/navigation/Header.tsx:4]: not familiar with react-icons/bi, did you have to add any yarn packages for this?
if possible, lets stick to` react-icons/md`, you can find icons here https://react-icons.github.io/react-icons/icons?name=md

**Review by gaoxk — CHANGES_REQUESTED** [review]
nicee love to see it! can you send screenshot to the design channel? it is so beautiful ✨

File: frontend/src/components/navigation/UserModal.tsx (hunk 1 above)
Comment (jennifer-lu) [reply frontend/src/components/navigation/UserModal.tsx:52]: Just to confirm, which props should I be alphabetizing? The UserModalProps or the ones in the tags?

Comment (jennifer-lu) [reply frontend/src/components/navigation/UserModal.tsx:52]: Nevermind I think I know, is it the import props?

**Comment by jennifer-lu** [comment]
By the way, is it normal for the yarn.lock file to have so many added lines? I'm not sure if I did something wrong.

File: frontend/src/components/navigation/UserModal.tsx (hunk 1 above)
Comment (gaoxk) [reply frontend/src/components/navigation/UserModal.tsx:52]: all the chakra ui props for these components, eg this is sorted alphabetically
```
 <Circle
              backgroundColor="white"
              height="60px"
              marginTop="35px"
              width="60px"
            >
```
(sorry - shouldve mentioned sorting!)

**Comment by gaoxk** [comment]
Leave the yarn.lock file in, we should pick up the habit of commiting it
https://stackoverflow.com/questions/39990017/should-i-commit-the-yarn-lock-file-and-what-is-it-for
its a big git diff because we havent updated it / committed it since last term LOL

**Review by gaoxk — APPROVED** [review]
feel free to merge after the prop order thing!

**Pull request merged**